modules/store/testing/src/mock_store.ts

```diff
@@ -27,12 +29,23 @@ export class MockStore<T> extends Store<T> {
     private state$: MockState<T>,
     actionsObserver: ActionsSubject,
     reducerManager: ReducerManager,
-    @Inject(INITIAL_STATE) private initialState: T
+    @Inject(INITIAL_STATE) private initialState: T,
+    @Inject(MOCK_SELECTORS) mockSelectors?: MockSelector[]
   ) {
     super(state$, actionsObserver, reducerManager);
     this.resetSelectors();
     this.state$.next(this.initialState);
     this.scannedActions$ = actionsObserver.asObservable();
+    if (mockSelectors) {
+      mockSelectors.forEach(mockSelector => {
+        const selector = mockSelector.selector;
+        if (typeof selector === 'string') {
```

Comment:
Any reason why this `if` statement is here? I think the `overrideSelector` is the same 😃

Reply:
Typescript doesn't seem to be able to infer between the two:

<img width="1018" alt="Screen Shot 2019-05-08 at 2 06 42 PM" src="https://user-images.githubusercontent.com/3779096/57399061-87492500-719e-11e9-9814-b806aef4b934.png">

Reply:
What if we would tweak the `overrideSelector` type?

```
overrideSelector<
    T,
    Result,
    SelectorKind =
      | string
      | MemoizedSelector<T, Result>
      | MemoizedSelectorWithProps<T, any, Result>
>(selector: SelectorKind, value: Result): SelectorKind;
```

Reply:
Tweaking the `overrideSelector` type straighten things out :
<img width="546" alt="Screen Shot 2019-05-09 at 5 00 53 PM" src="https://user-images.githubusercontent.com/3779096/57486427-11b38680-727c-11e9-89d2-f5d3a190979f.png">
However, it breaks the type checking of `overrideSelector` in the wild:
<img width="477" alt="Screen Shot 2019-05-09 at 5 02 25 PM" src="https://user-images.githubusercontent.com/3779096/57486628-6525d480-727c-11e9-8b0a-8666d7b5f117.png">
Should have a type error on the last line because `false` is not an applicable value for the selector.